pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -312,6 +315,34 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 			condemned = append(condemned, pods[i])
 		}
 		// If the ordinal could not be parsed (ord < 0), ignore the Pod.
+
+		for _, requested := range set.Spec.VolumeClaimTemplates {
+			pvcName := getPersistentVolumeClaimName(set, &requested, getOrdinal(pods[i]))
+			actual, err := ssc.client.CoreV1().PersistentVolumeClaims(set.Namespace).Get(pvcName, metav1.GetOptions{})
+			if err != nil {
```

Comment:
you also need to verfify if the err is of type NotFound

Reply:
Ah, right. Does it make sense to handle the other cases differently, though (except probably another log message)?